project/CONTRIBUTING.md

```diff
@@ -232,8 +232,62 @@ interface BarInterface
 }
 ```
 
-In some cases, you will have the possibility to warn the user that things will change,
-and recommend a new way of doing things. You can do so by triggering the dedicated kind of error, like this:
+##### Deprecation
+
+In some cases, you might have some code part being useless.
+If the concerned code is not tagged as `internal`, it must be deprecated on stable branch, then removed.
+
+If an alternate usage solution is possible, you **MUST** provide it on the deprecation message.
+
+The deprecated minor version **MUST NOT** be provided. Use `x` instead. It will be updated on releasing.
+
+Any deprecation **MUST** be documented on the correponding `UPGRADE-[0-9].x.md`.
+The documentation **MUST** be filled on the top **not** released section with a sub title.
+
+The `NEXT_MAJOR` tag should not be used for deprecation.
+The `@deprecated` and `E_USER_DEPRECATED` key will be matched for next major release.
```

Comment:
"matched for next major release" => "searched for before releasing the next major version"?